irods_browser_app/static/js/irods.js

```diff
@@ -0,0 +1,203 @@
+$('#btn-select-irods-file').on('click',function(){
+    $('#res_type').val($('#resource-type').val());
+    $('#file_struct').children().remove();
+	store = $('#root_store').val();
+	// Setting up the view tab
+	$('#file_struct').attr('name',store);
+	$('#irods_view_store').val(store);
+	// loading file structs
+	parent = $('#file_struct');
+	get_store(store,parent,0);
+	enable_settings();
+});
+
+function get_store(store,parent,margin){
+	margin_left = $(parent).css('margin-left');
+	margin_left = (parseInt(margin_left.substring(0,margin_left.length-2))-10) + 10;
+	form_data = new FormData();
+	form_data.append('store',store);
+	$.ajax({
+        mode: "queue",
+        url: '/irods/store/',
+        async: true,
+        type: "POST",
+        data: form_data,
+        processData: false,
+        contentType: false,
+        success: function (data, status) {
+            data = jQuery.parseJSON(data);
+            if (data[0].length ==0 && data[1].length ==0) {
+                $(parent).append("<div class='file' style='margin-left:"+margin+"px;'> -- EMPTY --</div>");
+            }
+            else {
+                $.each(data[1],function(i,v) {
+                    $(parent).append("<div class='folder' id = 'irods_folder_"+v+"' name='"+store+"/"+v+"' style='margin-left:"+margin+"px;'><img src='/static/img/folder.png' width='15' height='15'>&nbsp; "+v+"</div>");
+                });
+
+                $.each(data[0],function(i,v) {
+                    $(parent).append("<div class='file' id='irods_file_"+v+"' name='"+store+"/"+v+"' style='margin-left:"+margin+"px;'><img src='/static/img/file.png' width='15' height='15'>&nbsp; "+v+"</div>")
+                    $('.file').on('click',function(e){
+                        $('.file').css('background','');
+                        $('.file').removeClass('selected');
+                        $(e.target).css('background','#cecece');
+                        $(e.target).addClass('selected');
+                        set_datastore($(this).parent('div')[0], 0)
+                        return false
+                    });
+                });
+            }
+        },
+
+        error: function(status) {
+            return false;
+        }
+	});
+	return true;
+}
+
+function enable_settings(){
+	// Folder click settings
+    $('body').on('click', '.folder' ,function() {
+		margin_left = $(this).css('margin-left');
+		margin_left = parseInt(margin_left.substring(0,margin_left.length-2)) + 10;
+		if($(this).hasClass('isOpen')) {
+			$(this).addClass('isClose');
+			$(this).removeClass('isOpen');
+			$(this).children('div').hide();
+			set_datastore($(this).attr('name'),'folder');
+		}
+		else if($(this).hasClass('isClose')) {
+			$(this).addClass('isOpen');
+			$(this).removeClass('isClose');
+			$(this).children('div').show();
```

Comment:
This might be better to do with some CSS and a class you toggle to hide children?